src/aws_encryption_sdk/materials_managers/caching.py

```diff
@@ -111,10 +101,21 @@ def __attrs_post_init__(self):
         if self.max_age <= 0.0:
             raise ValueError("max_age cannot be less than or equal to 0")
 
+        options_provided = [
+            option is not None for option in (self.backing_materials_manager, self.keyring, self.master_key_provider)
+        ]
+        provided_count = len([is_set for is_set in options_provided if is_set])
+
+        if provided_count != 1:
+            raise TypeError("Exactly one of 'materials_manager', 'keyring', or 'key_provider' must be provided")
```

Comment:
Above you say that you prefer `backing_materials_manager`.
But here you error
and call it `materials_manager`